Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoMeraki.yaml

```diff
@@ -1,46 +1,580 @@
+                            (LogType == "events" and Substring has "Blocked DHCP server response"),
+                            "Blocked DHCP server response",
+                            LogType == "events" and (Substring has "VRRP packet" and not(Substring in~ ("VRRP passive", "VRRP active"))),
+                            "Virtual router collision",
+                            LogType in~ ("events", "airmarshal_events"),
+                            extract(@"type[ ]{0,1}[=]{0,1}[ ]{0,1}(\S+)", 1, Substring),
+                            ""
+                        )
+        | where (LogType in("flows", "firewall", "ids-alerts") or (LogType == "security_event" and LogSubType == "ids-alerted") or (LogType == "events" and (LogSubType in ("Blocked DHCP server response", "association", "Virtual router collision") or (LogSubType has "disassociation" and LogSubType !has "auth_neg_failed"))) or (LogType == "airmarshal_events" and LogSubType in("ssid_spoofing_detected", "rogue_ssid_detected")))
+        | parse-kv Substring as(timestamp: string) with (pair_delimiter=" ", kv_delimiter="=")
+        | extend 
+            Epoch = tostring(Parser[0]),
+            Device = tostring(Parser[1])
+        | extend
+            Epoch = iff(isnotempty(column_ifexists("timestamp", "")), timestamp, Epoch)
+        | extend
+            EpochTimestamp = split(Epoch, ".")
+        | extend
+            EventStartTime = unixtime_seconds_todatetime(tolong(EpochTimestamp[0]))
+        | parse-kv Substring as(src: string, ip_src: string, last_known_client_ip: string, ip: string, client_ip: string, dst: string, dns_server: string, client_mac: string, mac: string, shost: string, dhost: string, wired_mac: string, identity: string, sport: string, dport: string, direction: string, reason: string, protocol: string, priority: string, duration: string, vlan_id: string, signature: string, aid: string, rssi: string) with (pair_delimiter=" ", kv_delimiter="=")
+        | parse Substring with * " from " temp_addr " "restmessage
+        | extend temp_addr = iff(
+                            temp_addr has '"',
+                            replace_string(tostring(temp_addr), '"', ''),
+                            replace_string(tostring(temp_addr), "'", "")
+                        )
+        | extend SrcMacAddr = iff(temp_addr matches regex "(([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2}))", temp_addr, "")
+        | parse temp_addr with *  "[" temp_ip "]:" temp_port 
+        | extend SrcIpAddr = case(
+                            temp_addr has ".",
+                            split(temp_addr, ":")[0],
+                            isnotempty(temp_ip),
+                            temp_ip,
+                            temp_addr
+                        )
+        | extend SrcPortNumber = toint(
+                                case(
+        isnotempty(temp_port),
+        temp_port,
+        temp_addr has ".",
+        split(temp_addr, ":")[1],
+        ""
+    )
```

Comment:
Please beautify/format the document for better code readability.

Reply:
Done